src/storage/gcloud-storage/gcp-storage.service.ts

```diff
@@ -110,35 +118,47 @@ export class GCPStorageService {
         return ok(true);
     }
 
-    async verifyImageUrl(url: string): Promise<boolean> {
+    async verifyImageUrl(
+        url: string,
+    ): Promise<Result<boolean, ImageVerificationError>> {
         try {
             // Check if we're using the GCS emulator and verify the URL matches the emulator's base URL pattern
             if (this.emulatorHost) {
                 const emulatorBaseUrl = `${this.emulatorHost.replace('fake-gcs', 'localhost')}`;
-                return url.startsWith(emulatorBaseUrl);
+                if (!url.startsWith(emulatorBaseUrl)) {
+                    return error('invalid-url');
+                }
+                return ok(true);
             }
 
             // Verify if the URL matches the standard Google Cloud Storage public URL pattern for our bucket
             const gcsUrlPattern = new RegExp(
                 `https://storage.googleapis.com/${this.bucketName}/`,
```

Comment:
Why do we do a `startsWith` for the emulatorHost flow - but a regex test here?

Reply:
Not url is of type `URL` so I am directly checking host now